File: Python/BOJ_2506_점수계산.py
import sys

num = int(sys.stdin.readline())
judge = input().split(' ')

score = 0
count = 0
temp_score = 0
for i in judge:
    if i == '1':
        count += 1
    elif i == '0':
        for j in range(1, count + 1):
            temp_score += j
        score += temp_score
        count = 0
        temp_score = 0

for k in range(1, count + 1):
    temp_score += k
score += temp_score
print(score)

File: C/BOJ_2506_점수계산.c
#include <stdio.h>

int main()
{
	int num, score = 0, acc = 0;
	scanf("%d", &num);
	
	int arr[101];
	for (int i = 0; i < num; i++) scanf("%d", &arr[i]);
	for (int j = 0; j < num; j++) {
		if (arr[j] == 1) {
			acc++;
			score += acc;
		}
		if (arr[j] == 1 && arr[j + 1] == 0) acc = 0;
		
	}
	printf("%d", score);
	return 0;

}